### PR TITLE
zsh: new package at 5.7.1

### PR DESCRIPTION
**Pull request by dithpri**
## Checklist

- [x] Latest upstream version.
- [x] I agree to maintain this package (**required**).
- [x] I agree to notify KISS if I can no longer maintain this package (**required**).

### CODE REVIEW

File: community/zsh/patches/zsh_tests.patch [hunk 1]
```diff
@@ -0,0 +1,72 @@
+--- a/Test/A01grammar.ztst
++++ b/Test/A01grammar.ztst
+@@ -95,13 +95,14 @@
+ 0:`exec' with -l option
+ >-zsh
+ 
+-  (exec -a /bin/SPLATTER /bin/sh -c 'echo $0')
+-0:`exec' with -a option
+->/bin/SPLATTER
++# The following tests don't work if using busybox (symlinks)
```
Comment (dylanaraps) [inline community/zsh/patches/zsh_tests.patch]: Remove this patch.

File: community/zsh/build [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+    --enable-runhelpdir=/usr/share/zsh/help \
+    --enable-scriptdir=/usr/share/zsh/scripts \
+    --enable-additional-fpath=/usr/share/zsh/vendor-completions \
+    --enable-libc-musl \
+    --disable-gdbm \
+    --with-term-lib='ncursesw' \
+    --enable-cap
+
+make
+make test
```
Comment (dylanaraps) [inline community/zsh/build]: Remove `make test`

File: community/zsh/build [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+make DESTDIR="$1" install
```
Comment (dylanaraps) [inline community/zsh/build]: Add a new line below this.

**Comment by dithpri** [comment]
Done.

File: community/zsh/build [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+./configure \
+    --prefix=/usr \
+    --enable-function-subdirs \
+    --enable-fndir=/usr/share/zsh/functions \
+    --enable-runhelpdir=/usr/share/zsh/help \
+    --enable-scriptdir=/usr/share/zsh/scripts \
+    --enable-additional-fpath=/usr/share/zsh/vendor-completions \
+    --enable-libc-musl \
+    --disable-gdbm \
+    --with-term-lib='ncursesw' \
```
Comment (dylanaraps) [inline community/zsh/build]: Quotes here aren't needed.

File: community/zsh/sources [hunk 5]
```diff
@@ -0,0 +1,3 @@
+https://sourceforge.net/projects/zsh/files/zsh/5.7.1/zsh-5.7.1.tar.xz
+patches/remove_bin_duplicate.patch
+files/zprofile
```
Comment (dylanaraps) [inline community/zsh/sources]: Is shipping a `zprofile` file a typical part of packaging `zsh`?

File: community/zsh/depends [hunk 6]
```diff
@@ -0,0 +1,3 @@
+make make
```
Comment (dylanaraps) [inline community/zsh/depends]: `make` is an implicit dependencies and doesn't have to be listed.

File: community/zsh/depends [hunk 7]
```diff
@@ -0,0 +1,3 @@
...
+xz make
```
Comment (dylanaraps) [inline community/zsh/depends]: `xz` should go after `ncurses` (alphabetical ordering is desired though it sometimes makes sense to order things differently).

File: community/zsh/sources [hunk 8]
```diff
@@ -0,0 +1,3 @@
+https://sourceforge.net/projects/zsh/files/zsh/5.7.1/zsh-5.7.1.tar.xz
+patches/remove_bin_duplicate.patch
```
[the remark this answers is not in the document]
Comment (dithpri) [reply community/zsh/sources]: Both `zsh` and `zsh-5.7.1`, hardlinked. Seems this is used to perform a safe upgrade when zsh is installed directly from source without packaging.

File: community/zsh/sources (hunk 5 above)
Comment (dithpri) [reply community/zsh/sources]: It seems so. I know Gentoo, Arch and even Alpine provide a zprofile. The one I included will just include `/etc/profile`.

File: community/zsh/files/zprofile [hunk 9]
```diff
@@ -0,0 +1 @@
+emulate sh -c 'source /etc/profile'
```
[the remark this answers is not in the document]
Comment (dithpri) [reply community/zsh/files/zprofile]: Yup, I guess that's why it works. Didn't realize `source` was a bash-ism, thanks! Will change.

**Comment by dithpri** [comment]
If you have no further recommendations/objections, I'm ready to squash these commits before merging.

File: community/zsh/build [hunk 10]
```diff
@@ -0,0 +1,28 @@
+#!/bin/sh -e
+
+patch -p1 < remove_bin_duplicate.patch
+
+for _fpath in AIX BSD Cygwin Darwin Debian Mandriva openSUSE Redhat Solaris; do
+    rm -rf Completion/$_fpath
```
Comment (dylanaraps) [inline community/zsh/build]: Quote the argument with the variable.

File: community/zsh/build [hunk 11]
```diff
@@ -0,0 +1,28 @@
+#!/bin/sh -e
+
+patch -p1 < remove_bin_duplicate.patch
+
+for _fpath in AIX BSD Cygwin Darwin Debian Mandriva openSUSE Redhat Solaris; do
```
Comment (dylanaraps) [inline community/zsh/build]: Rename this to `path`.

File: community/zsh/sources (hunk 8 above)
[the remark this answers is not in the document]
Comment (dylanaraps) [reply community/zsh/sources]: I think it's better we include it. We already do so for any other package which provides a `-version` binary. There may also be a symlink configure option though I'm unsure.

File: community/zsh/build (hunk 10 above)
Comment (dithpri) [reply community/zsh/build]: That's not necessary in this case, but sure.

File: community/zsh/build [hunk 12]
```diff
@@ -0,0 +1,20 @@
+#!/bin/sh -e
+
+printf "functions='Completion/*comp* Completion/Base/*/* Completion/Linux/*/* Completion/Unix/*/* Completion/X/*/* Completion/Zsh/*/*'\n" >> Src/Zle/complete.mdd
```
Comment (dylanaraps) [inline community/zsh/build]: Something like this is also an option:

```sh
{
    printf "functions='"
    printf "Completion/*comp* "
    printf "Completion/Base/*/* "
    printf "Completion/Zsh/*/* "
    printf "Completion/Linux/*/* "
    printf "Completion/Unix/*/* "
    printf "Completion/X/*/* "
    printf "'\\n"
} >> Src/Zle/complete.mdd
```

This also makes adding/removing entries a little easier. Though a loop may turn out even simpler. ;)

Comment (dylanaraps) [reply community/zsh/build]: Another:

```sh
    printf '%s ' "functions='" "Completion/*comp*"    "Completion/Base/*/*" "Completion/Zsh/*/*" \
                               "Completion/Linux/*/*" "Completion/Unix/*/*" "Completion/X/*/*" "'
" >> Src/Zle/complete.mdd
```

Comment (dylanaraps) [reply community/zsh/build]: Or even:

```sh
    printf '%s ' "functions='" \
                 "Completion/*comp*"    \
                 "Completion/Base/*/*" \
                 "Completion/Zsh/*/*" \
                 "Completion/Linux/*/*" \
                 "Completion/Unix/*/*" \
                 "Completion/X/*/*" "'" >> Src/Zle/complete.mdd
```

Comment (dylanaraps) [reply community/zsh/build]: A loop: 

```sh
printf "functions='Completion/*comp*'\\n" >> Src/Zle/complete.mdd

for comp in Base Zsh Linux Unix X; do
    printf "functions+=' Completion/$comp/*/*'\\n" 
done >> Src/Zle/complete.mdd
```

(Untested)

Comment (dylanaraps) [reply community/zsh/build]: Another loop:

```sh
{
    printf "functions='Completion/*comp* " 

    for comp in Base Zsh Linux Unix X; do
        printf "Completion/$comp/*/* " 
    done >> Src/Zle/complete.mdd

    printf "'"
} >> Src/Zle/complete.mdd
```

**Pull request merged**